app/controllers/multifactor_auths_controller.rb

```diff
@@ -37,6 +43,50 @@ def update
     end
   end
 
+  def update
+    unless valid_mfa_level?
+      redirect_to edit_settings_path, flash: { error: t(".invalid_level") }
+      return
+    end
+
+    session[:level] = level_param
+    @user = current_user
+
+    setup_mfa_authentication
+    setup_webauthn_authentication
+
+    create_new_mfa_expiry
+
+    render template: "multifactor_auths/mfa_prompt"
+  end
+
+  def mfa_update
```

Comment:
This would be better as `totp_update`, as this PR would do the decoupling. For consistency with some other controllers, let's keep this as is this renaming could be done later (create a nice to have issue).